### ISSUE TITLE
How to implement custom coprocessor endpoints?

### ISSUE THREAD

**Issue by yuyang0** [tikv/tikv#6242]
## Question
Hello, I'm building a graph database based on tikv, the existing coprocessor implementation is not suitable for graph database, so I want to know if it is possiable to implement custom pluggable coprocessor endpoints,thx.

**Comment by sticnarf** [comment]
Yes, you can build and use your own query engine but need to modify tikv a bit. We don't have a plugin mechanism yet.

In the coprocessor endpoint, you can match your own request type in `parse_request` and build your own `RequestHandler`.
The existing request handlers should be a good reference. You can put your own query engine inside of tikv crate like `checksum.rs`. Or if you'd like to make it a separate crate, refer to `dag` and `tidb_query` crate.
